main.tf

```diff
@@ -1,5 +1,10 @@
 locals {
-  bucket_name                         = "${var.prefix}${random_id.uniq.hex}"
+
+  trimmed_bucket_arn = var.use_existing_bucket ? trimsuffix(var.bucket_arn, "/") : ""
+  bucket_arn         = var.use_existing_bucket ? local.trimmed_bucket_arn : aws_s3_bucket.eks_audit_log_bucket[0].arn
+  split_bucket_arn   = var.use_existing_bucket ? split(":", local.trimmed_bucket_arn) : []
+  bucket_name        = var.use_existing_bucket ? element(local.split_bucket_arn, (length(local.split_bucket_arn) - 1)) : "${var.prefix}-bucket-${random_id.uniq.hex}"
```

Comment:
Is this extracting the name from the bucket arn. Would it be possible to fetch the name with a data source rather than extracting from the arn input string. 

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/s3_bucket

Reply:
Talking with @djmctavish and the data source requires the bucket name. So extracting the name from the arn is necessary :(

Reply:
tried this using this but it requires the bucket name to create it, rather than supply an arn then the ability to extract the bucket name.